app/models/king.rb

```diff
@@ -1,12 +1,90 @@
 class King < Piece
-  # This method does not yet consider castling.
   def valid_move?(new_x, new_y)
     return false if off_board?(new_x, new_y)
     return false if current_square?(new_x, new_y)
 
     x_offset, y_offset = movement_by_axis(new_x, new_y)
-    return false unless (-1..1).cover?(x_offset) && (-1..1).cover?(y_offset)
+    return true if (-1..1).cover?(x_offset) && (-1..1).cover?(y_offset)
 
-    true
+    valid_castling?(new_x, new_y)
```

Comment:
Would `(-1..1).cover?(x_offset) && (-1..1).cover?(y_offset) ||  valid_castling?(new_x, new_y)` work?
The reason I say so is that `return true if...` is a smell